### PR TITLE
handle empty input #978

### PR DESCRIPTION
**Pull request by CaitlinOCallaghan**
- do not send empty query to service
- if input contains only spaces and line breaks, do not send to service, 
- if input contains only spaces and line breaks, clear input box
- only send inputs with textual content

### CODE REVIEW

File: src/client/features/enrichment/token-input.js [hunk 1]
```diff
@@ -56,7 +56,14 @@ class TokenInput extends React.Component {
           })
         ]),
         h('submit-container', {
-          onClick: () => { this.retrieveValidationAPIResult();} },
+          onClick: () => {
+            if(!inputBoxContents) return;
```
Comment (d2fong) [inline src/client/features/enrichment/token-input.js]: I think this logic should go in the retrieveValidationAPIResult code.

**Comment by d2fong** [comment]
Do you think it also makes sense to show a message to the user saying what was wrong with their input.

**Comment by maxkfranz** [comment]
I think the UI message makes sense, and we could add it in some future iteration.  Right now, time may be tight with testing and fixing.

File: src/client/features/enrichment/token-input.js [hunk 2]
```diff
@@ -24,6 +24,12 @@ class TokenInput extends React.Component {
     let { inputBoxContents } = this.state;
     let { controller } = this.props;
 
+    if(!inputBoxContents) return;
```
Comment (d2fong) [inline src/client/features/enrichment/token-input.js]: Why does inputBoxContents have to initially come from the parent props?

Comment (CaitlinOCallaghan) [reply src/client/features/enrichment/token-input.js]: That's an old part from before the refactoring. I just changed it now since its no longer needed.

Comment (d2fong) [reply src/client/features/enrichment/token-input.js]: Ok so you can remove this if statement now

File: src/client/features/enrichment/token-input.js [hunk 3]
```diff
@@ -24,12 +24,6 @@ class TokenInput extends React.Component {
     let { inputBoxContents } = this.state;
     let { controller } = this.props;
 
-    if(!inputBoxContents) return;
-    else if(!inputBoxContents.replace(/\s/g, "")) {
```
Comment (d2fong) [inline src/client/features/enrichment/token-input.js]: Sorry I meant that you only had to remove the first part of the else if.

You can return early if the replace check passes.

```js
if(!inputBoxContents.replace(/\s/g, "")) {
   this.setState({ inputBoxContents: "" });	
   return;
}
```

**Pull request merged**